instrument_view.php

```diff
@@ -20,16 +20,17 @@
 
 
 // Determine the request type (normal/csv/api).
+$disableAccessControl = isset( $disableAccessControl ) ? $disableAccessControl : false;
 $isApiRequest = isset( $isApiRequest ) ? $isApiRequest : false;
 $isInternalRequest = isset( $isInternalRequest ) ? $isInternalRequest : false;
 $isCsvDownload = ( ! $isApiRequest && isset( $_GET['download'] ) &&
-                   $module->isReportDownloadable( $reportID ) );
+                   ( $isInternalRequest || $module->isReportDownloadable( $reportID ) ) );
 $requestType = ( ( $isApiRequest && ! $isInternalRequest )
                  ? 'api' : ( $isCsvDownload ? 'download' : false ) );
 
 
 // Check user can view this report, redirect to main reports page if not.
```

Comment:
Am I getting a bit of deja-vu here?  These checks for request type and if the user can view it are repeated on all our instrument views.  Could it be consolidated in a function that all report types call?

Reply:
Possibly, although because of the various `_view.php` files being included in various functions to do the API/public/saveable features, there are potential issues with variable scope.